spec/sailfish_iptables_dbus_interface_description.xml

```diff
@@ -0,0 +1,464 @@
+"<!DOCTYPE node PUBLIC "-//freedesktop//DTD D-BUS Object Introspection 1.0//EN"
+"http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd">
+<node>
+
+<interface name="org.freedesktop.DBus.Introspectable">
+
+<method name="Introspect"><arg name="xml" type="s" direction="out"/>
+</method>
+
+</interface>
+
+<!-- The name of the sailfish connman iptables plugin interface -->
+<interface name="org.sailfishos.connman.mdm.iptables">
+
+<!-- Each function returns a code and descriptive string.
+The returned result codes and string pairs are:
+0	"Ok",
+1	"Invalid IP",
+2	"Invalid port",
+3	"Invalid port range",
+4	"Invalid service name",
+5	"Invalid protocol",
+6	"Invalid policy",
+7	"Invalid file path",
+8	"Cannot process rule",
+9	"Cannot perform operation",
+-->
+
+<!-- 
+Clears given table from iptables. Defaults to filter table.
+Sends signal FirewallCleared
+-->
+<method name="ClearIptablesTable">
+<arg name="table" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!--
+Change policy of filter table INPUT chain. Parameter is the same as iptables
+command (ACCEPT/DROP)
+Sends signal PolicyChanged
+-->
+<method name="ChangeInputPolicy">
+<arg name="policy" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!--
+Change policy of filter table OUTPUT chain. Parameter is the same as iptables
+command (ACCEPT/DROP)
+Sends signal PolicyChanged
+-->
+<method name="ChangeOutputPolicy">
+<arg name="policy" type="s" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!--
+Following methods take following parameters as input:
+ NAME		TYPE	DESCRIPTION
+ ip		s		IP address or network with CIDR or dot notation (1.2.3.4/24
+ 					or 1.2.3.4/255.255.255.0)
+ port		q		Port number as unsigned 16bit integer
+ port		s		Port as string, single port or port range, separator is ":"
+ service	s		Service name, case insensitive, 
+ operation	s 		Perform operation add/remove, case insensitive, defaults to
+ 					add if empty string. Add appends the rule to iptables.
+ 					Remove removes the rule from iptables.
```

Comment:
String? Case sensitive or not? Are leading/trailing spaces ignored or not? Just use a number, please. Numbers are so much easier to deal with.

Reply:
This is not critical though. Feel free to ignore.

Reply:
See next comment. In convenience API methods I think I'll use integers to replace these strings.

Reply:
Changed operation to q (unsigned 16bit int) in 4df4a6dde2b8da11000867ed3a69aecc60ce3eee